src/App.jsx

```diff
@@ -83,42 +65,37 @@ function App() {
   };
 
   return (
-    <LanguageContext.Provider value={language}>
-      <Header
-        language={language}
-        onLanguageChanged={(language) => setLanguage(language)}
-      />
-
+    <>
+      <Header />
       <main className={cx.main}>
         {viewState.name === "Dashboard" && (
-          <Dashboard
-            entries={entries}
-            onEntryEdit={goToEditEntry}
-            onEntryDelete={handleEntryDeleted}
-            onEntryNew={goToNewEntry}
-          />
+          <Dashboard entries={entries} onEntryDelete={handleEntryDeleted} />
         )}
 
         {viewState.name === "New Entry" && (
-          <NewEntryForm
-            onSubmit={handleNewEntrySubmitted}
-            goToDashboard={goToDashboard}
-          />
+          <NewEntryForm onSubmit={handleNewEntrySubmitted} />
         )}
 
         {viewState.name === "Edit Entry" && (
           <EditEntryForm
             entry={entries.find((entry) => entry.id === viewState.id)}
-            goToDashboard={goToDashboard}
             onEntryDelete={() => handleEntryDeleted(viewState.id)}
             onSubmit={(entryIntent) =>
               handleEntryEdited(viewState.id, entryIntent)
             }
           />
         )}
       </main>
-    </LanguageContext.Provider>
+    </>
   );
 }
```

Comment:
the ideal scenario would be putting all this logic in a separate component and leaving `App` only with what you return in `AppWithProviders()`